src/archives.c

```diff
@@ -0,0 +1,155 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2012-2017 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *   Authors:
+ *         Matthew Johnson <mathew.johnson@intel.com>
+ *
+ */
+
+#include <archive.h>
+#include <archive_entry.h>
+#include <stdlib.h>
+
+/* archive_check_err(ar, ret)
+ *
+ * Check and print archive errors */
+static int archive_check_err(struct archive *ar, int ret)
+{
+	if (ret < ARCHIVE_OK) {
+		/* print error */
+		fprintf(stderr, "%s\n", archive_error_string(ar));
+	}
+
+	if (ret < ARCHIVE_WARN) {
+		/* error was worse than a warning */
+		return 1;
+	}
+
+	/* just a warning */
+	return 0;
+}
+
+/* copy_data(ar, aw)
+ *
+ * Copy archive data from ar to aw */
+static int copy_data(struct archive *ar, struct archive *aw)
+{
+	int r;
+	const void *buffer;
+	size_t size;
+	la_int64_t offset;
```

Comment:
Just to be pedantic, from the man pages the function below is defined  archive_read_data_block(struct archive *, const void **buff, size_t *len, off_t *offset); 
I know off_t 'cause it's a standard type, so it would make sense to match them up and make it: off_t offset IMHO, then all the types are obvious and matching.

Reply:
Done!